grails-app/services/org/pih/warehouse/report/ReportService.groovy

```diff
@@ -760,15 +779,10 @@ class ReportService implements ApplicationContextAware {
 
             def results = dataService.executeQuery(query, params)
             data = results.collect {
```

Comment:
If you had to pull out the join into a separate query to fix the data, then instead of calling the second method for each product separately in the ReportController while collecting data you could do it here:

1. Remove product code and product name from select here - return only product ids (you are pulling the product in the controller anyway).
2. Instead of doing the collect here, call the getOnOrderData here and pass the list of product ids into it. 
3. In the getOnOrderData instead of doing `where product_id = :productId` do `where product_id in (:productIds)`.
4. Add a product id into the selected data in the getOnOrderData and return it here.
5. Do some small adjustments in the report controller according to the points above.

Reply:
ok, thanks.

Reply:
done